Extension is trying to use npm instead of cargo?

Hi there,

When I run `cargo tauri dev` in the Terminal, my app starts correctly, but when I try using `Tauri: Dev` from the Command Palette, I get this output:

```
npm ERR! Missing script: "tauri"
npm ERR! 
npm ERR! To see a list of scripts, run:
npm ERR!   npm run

npm ERR! A complete log of this run can be found in: /Users/andrew/.npm/_logs/2024-01-22T17_16_56_539Z-debug-0.log
-----------------------
```

Is there a way to tell it to use cargo instead?

--- a/.changes/fix-cargo.md
+++ b/.changes/fix-cargo.md
@@ -1,0 +1,5 @@
+---
+'tauri-vscode': patch
+---
+
+Fix extension trying to use npm instead of cargo.

--- a/src/extension.ts
+++ b/src/extension.ts
@@ -413,18 +413,13 @@ function __getNpmCommand() {
 }
 
 function __getPackageManagerCommand(projectPath: string): string | null {
+  const isNodeProject = __isNodeProject(projectPath)
+  if (!isNodeProject) return 'cargo'
 
-  try {
-    const packageJson = JSON.parse(
-      fs.readFileSync(`${projectPath}/package.json`, 'utf8')
-    )
-
-    // if there is no tauri command, use cargo
-    if (!packageJson.script || !packageJson.script['tauri']) return 'cargo'
-  } catch {
-    // if project uses vanilla js i.e. if no package.json exists
-    return 'cargo'
-  }
+  const packageJson = JSON.parse(
+    fs.readFileSync(`${projectPath}/package.json`, 'utf8')
+  )
+  if (!packageJson.script || !packageJson.script['tauri']) return 'cargo'
 
   const m = __usePnpm(projectPath)
     ? 'pnpm'

--- a/src/extension.ts
+++ b/src/extension.ts
@@ -413,6 +413,19 @@ function __getNpmCommand() {
 }
 
 function __getPackageManagerCommand(projectPath: string): string | null {
+
+  try {
+    const packageJson = JSON.parse(
+      fs.readFileSync(`${projectPath}/package.json`, 'utf8')
+    )
+
+    // if there is no tauri command, use cargo
+    if (!packageJson.script || !packageJson.script['tauri']) return 'cargo'
+  } catch {
+    // if project uses vanilla js i.e. if no package.json exists
+    return 'cargo'
+  }
+
   const m = __usePnpm(projectPath)
     ? 'pnpm'
     : __useYarn(projectPath)

--- a/src/extension.ts
+++ b/src/extension.ts
@@ -414,30 +414,29 @@ function __getNpmCommand() {
 
 function __getPackageManagerCommand(projectPath: string): string | null {
   const isNodeProject = __isNodeProject(projectPath)
-  if (!isNodeProject) return 'cargo'
+  if (isNodeProject) {
+    if (__usePnpm(projectPath)) return 'pnpm'
+    if (__useYarn(projectPath)) return 'yarn'
 
-  const packageJson = JSON.parse(
-    fs.readFileSync(`${projectPath}/package.json`, 'utf8')
-  )
-  if (!packageJson.script || !packageJson.script['tauri']) return 'cargo'
-
-  const m = __usePnpm(projectPath)
-    ? 'pnpm'
-    : __useYarn(projectPath)
-      ? 'yarn'
-      : __useNpm(projectPath)
-        ? __getNpmCommand()
-        : __useCargo()
-          ? 'cargo'
-          : null
-
-  if (!m) {
-    vscode.window.showErrorMessage(
-      "Couldn't detect package manager for current project. Try running Tauri: Init Command"
+    const packageJson = JSON.parse(
+      fs.readFileSync(`${projectPath}/package.json`, 'utf8')
     )
+
+    if (
+      __useNpm(projectPath) &&
+      packageJson.script &&
+      packageJson.script['tauri']
+    )
+      return __getNpmCommand()
   }
 
-  return m
+  if (__useCargo()) return 'cargo'
+
+  vscode.window.showErrorMessage(
+    "Couldn't detect package manager for current project. Try running Tauri: Init Command"
+  )
+
+  return null
 }
 
 interface RunOptions {

